editor/src/clj/dynamo/graph.clj

```diff
@@ -1156,8 +1172,13 @@
                                         (map #(vector % serial-id)
                                              (override-originals basis original-id))))
                               replacements)
-         include-arc?   (partial ig/arc-endpoints-p (partial contains? serial-ids))
-         serialize-arc  (partial serialize-arc serial-ids)
+         include-arc?   (partial ig/arc-endpoints-p
+                                 (fn [id label]
+                                   (or (contains? serial-ids id)
+                                       (and (contains? external-refs id)
+                                            (contains? (get external-labels id) label)))))
+         arc-ids        (into serial-ids external-refs)
```

Comment:
Maybe `serial-ids-for-arcs` would be clearer?

Reply:
Not sure, since it's no longer serial ids, but a merge of copied-node-id->serial-id with external-node-id->external-ref-keyword. Maybe `serialize-dictionary` to mirror `deserialize-dictionary` in `paste`?

Reply:
Yes, that makes sense!